ansys/mapdl/core/post.py

```diff
@@ -289,7 +289,9 @@ def nodal_values(self, item, comp="") -> np.ndarray:
 
         """
         # using _ndof_rst instead of get_array because it is wrapped to check the rst.
-        return self._ndof_rst(self, item, it1num=comp)
+        values = self._ndof_rst(item=item, it1num=comp)
+        mask = self.selected_nodes
+        return values[mask]
```

Comment:
There's no reason to place this on three lines when one line will do provide it's readable.
```suggestion
        return self._ndof_rst(item=item, it1num=comp)[self.selected_nodes]
```

Reply:
This makes sense.